File: ios/src/RNiOSOggDecoderModule.swift
import ExpoModulesCore

public class RNiOSOggDecoderModule: Module {
    public func definition() -> ModuleDefinition {
        Name("RNiOSOggDecoder")
        
        
        Function("decode") { (sourcePath: String, destPath: String) in
            let decoder = OggDecoder()
            decoder.decode(sourcePath, destPath: destPath)
            return destPath
        }
        
    }
}

File: src/RNiOSOggDecoderModule.ts
import { NativeModule, requireNativeModule } from "expo";

declare class RNiOSOggDecoderModule extends NativeModule {
  decode(sourcePath: string, destPath: string): Promise<string | undefined>;
}

// This call loads the native module object from the JSI.
export default requireNativeModule<RNiOSOggDecoderModule>("RNiOSOggDecoder");
